kern/postgres_kern.c

```diff
@@ -0,0 +1,42 @@
+#include "ecapture.h"
+#include "common.h"
+
+struct data_t {
+    u64 pid;
+    u64 timestamp;
+    char query[MAX_DATA_SIZE_POSTGRES];
+    char comm[TASK_COMM_LEN];
+};
+
+struct
+{
+    __uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+// https://github.com/postgres/postgres/blob/7b7ed046cb2ad9f6efac90380757d5977f0f563f/src/backend/tcop/postgres.c#L987-L992
+// hook function exec_simple_query
+// versions 10 - now
+// static void exec_simple_query(const char *query_string)
+SEC("uprobe/exec_simple_query")
+int postgres_query(struct pt_regs *ctx) {
```

Comment:
what is the difference between `exec_simple_query` and `exec_parse_message`, Why not `exec_parse_message`?

Reply:
exec_parse_message called in exec_simple_query.